### PR TITLE
Codegen is deterministic sans formatting

### PR DESCRIPTION
**Pull request by tjb9dc**
Use ordered sets in the in the AST.

I opted to use the library over rolling our own (e.g. wrapping dict keyset) https://github.com/rspeer/ordered-set

I duplicated the seed tests in the fastapi workspace and added an entry with `skip_formatting` custom config. I reran seed a few times and it produced no diff on the unformatted code! Then I tried reverting my ordered set change, and got diffs! This leads me to believe that we got ordered sets in the places that count.

### CODE REVIEW

**Comment by tjb9dc** [comment]
@dsinghvi if yall are open to going this route, we should add some `seed` tests that cover the determinism with `skip_formatting` enabled

**Comment by dsinghvi** [comment]
> @dsinghvi if yall are open to going this route, we should add some `seed` tests that cover the determinism with `skip_formatting` enabled

sounds great

File: poetry.lock [hunk 1]
```diff
@@ -1,9 +1,10 @@
-# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
+# This file is automatically @generated by Poetry and should not be changed by hand.
 
 [[package]]
 name = "anyio"
 version = "3.6.2"
 description = "High level compatibility layer for multiple asynchronous event loop implementations"
+category = "main"
```
Comment (tjb9dc) [inline poetry.lock:7]: Not sure how we feel about all this category stuff. I'm gonna prune this diff I think it's due to my poetry env being different

Comment (dsinghvi) [reply poetry.lock:7]: awesome, yeah i think we should ideally remove

Comment (tjb9dc) [reply poetry.lock]: alrighty @dsinghvi should be good now!

**Pull request merged**